test/es6/dynamic-module-import-specifier.js

```diff
@@ -0,0 +1,142 @@
+        thenFunc(v);
+        _asyncExit();
+    }).catch((err)=>{
+        _asyncEnter();
+        catchFunc(err);
+        _asyncExit();
+    });
+}
+
+var tests = [
+    {
+        name: "Valid cases for import()",
+        body: function () {
+                let functionBody =
+                    `
+                    assert.doesNotThrow(function () { eval("import(undefined)"); }, "undefined");
+                    assert.doesNotThrow(function () { eval("import(null)"); }, "null");
+                    assert.doesNotThrow(function () { eval("import(true)"); }, "boolean - true");
+                    assert.doesNotThrow(function () { eval("import(false)"); }, "boolean - false");
+                    assert.doesNotThrow(function () { eval("import(1234567890)"); }, "number");
+                    assert.doesNotThrow(function () { eval("import('abc789cde')"); }, "string literal");
+                    assert.doesNotThrow(function () { eval("import('number' + 100 + 0.4 * 18)"); }, "expression");
+                    assert.doesNotThrow(function () { eval("import(import(true))"); }, "nested import");
+                    assert.doesNotThrow(function () { eval("import(import(Infinity) + import(undefined))"); }, "nested import expression");
+                    `;
+
+                testScript(functionBody, "Test importing a simple exported function");
+                testModuleScript(functionBody, "Test importing a simple exported function");
+        }
+    },
+    {
+        name: "Syntax errors for import() call",
+        body: function () {
+                let functionBody =
+                    `
+                    assert.throws(function () { eval("import()"); }, SyntaxError, "no argument");
+                    assert.throws(function () { eval("import(1, 2)"); }, SyntaxError, "more than one arguments");
+                    assert.throws(function () { eval("import('abc.js', 'def.js')"); }, SyntaxError, "more than one argument - case 2");
+                    assert.throws(function () { eval("import(...['abc.js', 'def.js'])"); }, SyntaxError, "spread argument");
+                    `;
+
+                testScript(functionBody, "Test importing a simple exported function");
+                testModuleScript(functionBody, "Test importing a simple exported function");
+        }
+    },
+    {
+        name: "Module specifier that are not string",
+        body: function () {
+            var testNonStringSpecifier = function (specifier, expectedType, expectedErrMsg, message) {
+                if (typeof message === "undefined" ) {
+                    message = specifier;
+                }
+
+                let functionBody =
+                    `testDynamicImport(
+                        ()=>{
+                            return import(${specifier});
+                        },
+                        (v)=>{
+                            assert.isTrue(false, 'Expected: promise rejected; actual: promise resolved: ' + '${message}');
```

Comment:
Done.

---
In reply to: [115364860](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115364860) [](ancestors = 115364860)